### PR TITLE
Docs: Add Example Subsections

### PR DESCRIPTION
**Pull request by ax3l**
Add subsections to our example section.

### CODE REVIEW

File: docs/source/usage/examples.rst [hunk 1]
```diff
@@ -3,41 +3,70 @@
...
 
 
+Virtual Test Stands
+-------------------
+
+.. toctree::
+   :maxdepth: 1
+
+   examples/cyclotron/README.rst
+   examples/iota_lens/README.rst
```
Comment (ax3l) [inline docs/source/usage/examples.rst]: Maybe the lens can go to earlier sections.
```suggestion
```

Comment (cemitch99) [reply docs/source/usage/examples.rst]: Yes, agreed.  I think the lens should go in the single particle dynamics section.

File: docs/source/usage/examples.rst [hunk 2]
```diff
@@ -3,41 +3,70 @@
...
+Space Charge
+------------
+
+.. toctree::
+   :maxdepth: 1
+
+   examples/expanding_beam/README.rst
+   examples/cfchannel/README.rst
+   examples/kurth/README.rst
+
```
Comment (cemitch99) [inline docs/source/usage/examples.rst:46]:
```suggestion
   examples/epac2004_benchmarks/README.rst

```

**Review by cemitch99 — COMMENTED** [review]
Thanks; I think these are good categories.  I might add a category called Distribution Initialization (or similar), and put `examples/distgen`.  I noticed that the README appears to be missing in that subdirectory, so I will add one.  If the list of single particle dynamics examples becomes too long, we might think about splitting it.

File: docs/source/usage/examples.rst (hunk 2 above)
Comment (ax3l) [reply docs/source/usage/examples.rst:46]: Oh no, how did I drop this

File: docs/source/usage/examples.rst [hunk 3]
```diff
@@ -3,41 +3,70 @@
...
-   examples/rotation/README.rst
    examples/solenoid_softedge/README.rst
    examples/quadrupole_softedge/README.rst
-   examples/positron_channel/README.rst
-   examples/cyclotron/README.rst
    examples/cfbend/README.rst
    examples/compression/README.rst
    examples/kicker/README.rst
    examples/thin_dipole/README.rst
    examples/aperture/README.rst
```
Comment (ax3l) [inline docs/source/usage/examples.rst:31]:
```suggestion
   examples/aperture/README.rst
   examples/iota_lens/README.rst
```

**Review by cemitch99 — APPROVED** [review]
This looks good to me.  I can add the distgen README + Distribution category as a follow-up.

**Pull request merged**